### PR TITLE
Fix wic numbering

### LINKED ISSUE #1011: Wic numbering error on 3745 and 3725
https://gns3.com/qa/gns3-2-0-3725-3745-wics-numberin

image

image

### PR DESCRIPTION
**Pull request by noplay**
Fix #1011

### CODE REVIEW

**Comment by grossmj** [comment]
Does not work when connecting 2 routers together with WIC interfaces:

Get errors like this:

```
Error while creating link: Port 21 does not exist in adapter GT96100-FE
Error while creating link: Port 17 does not exist in adapter GT96100-FE
```

Also, not all interfaces are displayed in the tooltip when mouse hovering a router.

**Comment by noplay** [comment]
Which router image are you using?  I don't reproduce the error

File: gns3server/controller/ports/port_factory.py [hunk 1]
```diff
@@ -194,7 +194,9 @@ def __new__(cls, properties):
                 port_class = cls.WIC_MATRIX[properties[name]]["port"]
                 if port_class:
                     for port_number in range(0, cls.WIC_MATRIX[properties[name]]["nb_ports"]):
-                        name = "{}{}/{}".format(port_class.long_name_type(), 0, wic_port_number)
-                        ports.append(port_class(name, 0, 0, wic_port_number))
+                        name = "{}{}/{}".format(port_class.long_name_type(), 0, wic_port_number - 16)
```
Comment (grossmj) [inline gns3server/controller/ports/port_factory.py]: Should be something like wic_port_number - (wic_port_number * 16).

File: gns3server/controller/ports/port_factory.py [hunk 2]
```diff
@@ -194,7 +194,9 @@ def __new__(cls, properties):
...
+                        port = port_class(name, 0, 0, wic_port_number)
+                        port.short_name = "{}{}/{}".format(port.short_name_type, 0, wic_port_number - 16)
```
Comment (grossmj) [inline gns3server/controller/ports/port_factory.py]: Should be something like wic_port_number - (wic_port_number * 16)

**Comment by grossmj** [comment]
Testing with c3725-adventerprisek9-mz.124-15.T14.image.

The first WIC slot works but not the others (try to connect s0/4 for instance).

You must account for slot1 and slot2. Internally Slot1 interfaces start at 32 and Slot2 interfaces start at 48.

File: gns3server/controller/ports/port_factory.py (hunk 1 above)
Comment (grossmj) [inline gns3server/controller/ports/port_factory.py]: Actually wic_slot_number * 16...

File: gns3server/controller/ports/port_factory.py (hunk 2 above)
Comment (grossmj) [inline gns3server/controller/ports/port_factory.py]: Actually wic_slot_number * 16...

**Comment by noplay** [comment]
Fixed could you confirm, I'm right now?

**Comment by grossmj** [comment]
All good now.

**Pull request merged**